here-naksha-lib-plv8/src/commonMain/kotlin/NakshaFeaturesWriter.kt

```diff
@@ -83,10 +106,21 @@ class NakshaBulkLoader(
                         addDelStmt(plan.insertDelPlan(), featureRowMap)
                         addDeleteHeadStmt(plan.deleteHeadPlan(), featureRowMap)
                         if (isHistoryDisabled == false) addCopyDelToHstStmt(plan.copyDelToHstPlan(), featureRowMap)
+                        if (!minResult && opType == XYZ_OP_DELETE) {
```

Comment:
Why do we need to select the row, shouldn't we have it already? When the methods starts, we select all existing rows for update to acquire a lock on the rows, there we always read the rows, when we keep them in a hash map, we can just read them here without any db query?

Reply:
we don't fetch feature:bytea, geo:bytea etc. when we acquire lock, so if we call delete feature (by id) we have to fetch it.

Reply:
Good point, but how about fetching the full feature upfront if the operation is delete? You could combine two CTEs like (simplified a lot) via `WITH small AS (SELECT ... WHERE ), full AS (SELECT ... WHERE) select * from small, full ...`?